docs/source/examples/managed-jobs.rst

```diff
@@ -282,6 +282,28 @@ candidate resources for a job. See documentation :ref:`here
 In this example, SkyPilot will perform cost optimizations to select the resource to use, which almost certainly
 will be spot instances. If spot instances are not available, SkyPilot will fall back to launch on-demand instances.
 
+
+Advanced Strategies for Recovery
+---------------------------------
+
+Recovery on User Code Crash/Failure
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
How about keep this simple in 1 section (along with L304)? Maybe "Job Restarts on User Code Failure"?